stdlib/importlib/metadata/_meta.pyi

```diff
@@ -8,15 +9,34 @@ class PackageMetadata(Protocol):
     def __contains__(self, item: str) -> bool: ...
     def __getitem__(self, key: str) -> str: ...
     def __iter__(self) -> Iterator[str]: ...
-    def get_all(self, name: str, failobj: _T = ...) -> list[Any] | _T: ...
     @property
     def json(self) -> dict[str, str | list[str]]: ...
+    if sys.version_info >= (3, 12):
+        @overload
+        def get(self, name: str, failobj: None = None) -> str | None: ...
+        @overload
+        def get(self, name: str, failobj: _T) -> _T | str: ...
+        @overload
+        def get_all(self, name: str, failobj: None = None) -> list[Any] | None: ...
+        @overload
+        def get_all(self, name: str, failobj: _T) -> list[Any] | _T: ...
+    else:
+        def get_all(self, name: str, failobj: _T = ...) -> list[Any] | _T: ...
```

Comment:
The TypeVar could go unsolved here if no value is passed for `failobj`, which would lead the return type being inferred as `list[Any] | Any` (I know this is a problem in the existing annotations we have here, but that doesn't mean that we shouldn't fix it ;)

It's likely that we should just make it an overloaded function on Python <3.12 as well. I don't think the contract for `get_all()` changed in Python 3.12, it's just that the `importlib_metadata` authors learned a few new `typing` tricks and figured out how to improve their annotations